libs/common/include/launchdarkly/error.hpp

```diff
@@ -27,7 +26,11 @@ enum class Error : std::uint32_t {
     kMax = std::numeric_limits<std::uint32_t>::max()
 };
 
-char const* ErrorToString(Error err);
```

Comment:
Made `Error` be a variant, and renamed the existing `Error` (enum) to `ErrorCode`. 

I also added a `bool operator==(Error, ErrorCode)` to satisfy existing tests.

Reply:
I am confused. LDStatus is Opaque, but Error, in C++ land, was not?